openthread/cmake/extensions.cmake

```diff
@@ -3,20 +3,44 @@
 #
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
+include(${NRFXLIB_DIR}/common.cmake)
+include(${NRFXLIB_DIR}/nrf_security/cmake/extensions.cmake)
+
+# The Build the path of the library taking in nrf_security backend type,
+# OpenThread version and feature set  
+function(openthread_calculate_lib_path ot_version lib_path)
+  if(CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER)
```

Comment:
These three defines(`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER`, `CONFIG_OPENTHREAD_NORDIC_LIBRARY_FTD`, `CONFIG_OPENTHREAD_NORDIC_LIBRARY_MTD`) are basically used to pick the source/destination library folder depending if we are building from source or libraries.

I think that we should keep in mind to expand the `ug_thread_configuring.rst` as it currently states:
```
* :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER` - Enable the complete set of OpenThread features.
* :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_FTD` - Enable optimized OpenThread features for FTD.
* :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MTD` - Enable optimized OpenThread features for MTD.
```
Which is true only for building from library. When building libraries user will just overwrite selected libraries with whatever configuration he already has. I think that while reading this documentation one may think that it would also preconfigure the stack with features appropriate for selected configuration.

Reply:
Any related documentation will be updated in reference sdk-nrf pull request.
Currently install_openthread_library target is active only if CONFIG_OPENTHREAD_NORDIC_LIBRARY which imply one of possible option: master, ftd, mtd. If we want a user to create library with custom configuration we could add another option there. What to you think?

Reply:
It makes sense. With the current approach he would be forced to overwrite one of existing configurations.